nautobot/utilities/tables.py

```diff
@@ -317,3 +330,39 @@ def render(self, value):
             trunc = Truncator(value)
             value = trunc.words(self.truncate_words)
         return value
+
+
+class CustomFieldColumn(tables.Column):
+    """
+    Display custom fields in the appropriate format.
+    """
+
+    def __init__(self, customfield, *args, **kwargs):
+        self.customfield = customfield
+        kwargs["accessor"] = Accessor(f"custom_field_data__{customfield.name}")
+        kwargs["verbose_name"] = customfield.label or customfield.name
+
+        super().__init__(*args, **kwargs)
+
+    def render(self, record, bound_column, value):
+        if value is None:
+            return self.default
+
+        if self.customfield.type == CustomFieldTypeChoices.TYPE_BOOLEAN:
+            if value is True:
+                template = '<span class="text-success"><i class="mdi mdi-check-bold"></i></span>'
+            else:
+                template = '<span class="text-danger"><i class="mdi mdi-close-thick"></i></span>'
+        elif self.customfield.type == CustomFieldTypeChoices.TYPE_MULTISELECT:
+            if value:
+                template = ""
+                for v in value:
+                    template += f'<span class="label label-default">{v}</span> '
```

Comment:
Please use `format_html` throughout so that potentially-HTML-unsafe custom field values automatically get properly escaped.

```suggestion
                    template += format_html('<span class="label label-default">{}</span> ', v)
```